c++/include/orc/Vector.hh

```diff
@@ -248,10 +249,15 @@ namespace orc {
     ~EncodedStringVectorBatch() override;
     std::string toString() const override;
     void resize(uint64_t capacity) override;
+
+    // Calculate data and length in StringVectorBatch from dictionary and index
+    void calculateDataAndLength();
+
     std::shared_ptr<StringDictionary> dictionary;
 
     // index for dictionary entry
     DataBuffer<int64_t> index;
+    std::atomic<bool> calculated{false};
```

Comment:
Do we need atomic type here?

Reply:
It is needed to avoid duplicated invokes from caller.

Reply:
Sorry for the late comment. All members of the ColumnVectorBatch class are non-atomic. It is assumed that the ColumnVectorBatch object will not be shared or reused across multiple threads. Could you share the code in ClickHouse that invokes decodeDictionary()?